### ISSUE TITLE
Molecule "api-test" failure in graph generation on ossm 2.1

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#4246]
Test is looking for two nodes, but there are 4 nodes.

This runs by installing the latest quay image of Kiali using the helm installer.

This test passes on upstream Istio 1.10 on minikube and Istio 1.11.rc2 on OpenShift 4.8.
This fails on OSSM 2.1 on OpenShift 4.7.

```
    TASK [Results: /namespaces/graph: https://kiali-istio-system.apps.ocp4.local/api/namespaces/graph?namespaces=simple-mesh&graphType=workload&injectServiceNodes=false] ***
    task path: /tmp/molecule/operator/molecule/api-test/graph.test.yaml:16
    ok: [localhost] => {
        "msg": {
            "duration": 600,
            "elements": {
                "edges": [
                    {
                        "data": {
                            "destPrincipal": "spiffe://cluster.local/ns/simple-mesh/sa/simple-sa",
                            "id": "cea90af87bc1deaf1d705a1afa31d582",
                            "isMTLS": "100",
                            "responseTime": "19",
                            "source": "3441ac506bb6b7b88ba0de131d9b3c72",
                            "sourcePrincipal": "spiffe://cluster.local/ns/simple-mesh/sa/simple-sa",
                            "target": "b839f456761ee639d46c1ae31dfc3cf2",
                            "throughput": "739",
                            "traffic": {
                                "protocol": "http",
                                "rates": {
                                    "http": "0.71",
                                    "httpPercentReq": "100.0"
                                },
                                "responses": {
                                    "200": {
                                        "flags": {
                                            "-": "100.0"
                                        },
                                        "hosts": {
                                            "simple-server-svc.simple-mesh.svc.cluster.local": "100.0"
                                        }
                                    }
                                }
                            }
                        }
                    }
                ],
                "nodes": [
                    {
                        "data": {
                            "app": "simple-client",
                            "cluster": "Kubernetes",
                            "id": "371ae5a8eff043855792872141efa353",
                            "isIdle": true,
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "version": "v99",
                            "workload": "simple-client"
                        }
                    },
                    {
                        "data": {
                            "app": "simple-server",
                            "cluster": "Kubernetes",
                            "id": "440df82a974d31fc2ab0e4b7ba8bc1ed",
                            "isIdle": true,
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "version": "v99",
                            "workload": "simple-server"
                        }
                    },
                    {
                        "data": {
                            "app": "simple-client",
                            "cluster": "unknown",
                            "id": "3441ac506bb6b7b88ba0de131d9b3c72",
                            "isRoot": true,
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "traffic": [
                                {
                                    "protocol": "http",
                                    "rates": {
                                        "httpOut": "0.71"
                                    }
                                }
                            ],
                            "version": "v99",
                            "workload": "simple-client"
                        }
                    },
                    {
                        "data": {
                            "app": "simple-server",
                            "cluster": "unknown",
                            "destServices": [
                                {
                                    "cluster": "unknown",
                                    "name": "simple-server-svc",
                                    "namespace": "simple-mesh"
                                }
                            ],
                            "id": "b839f456761ee639d46c1ae31dfc3cf2",
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "traffic": [
                                {
                                    "protocol": "http",
                                    "rates": {
                                        "httpIn": "0.71"
                                    }
                                }
                            ],
                            "version": "v99",
                            "workload": "simple-server"
                        }
                    }
                ]
            },
            "graphType": "workload",
            "timestamp": 1627886741
        }
    }
    
    TASK [Assert that we were able to get the graph info for the small 2-node mesh] ***
    task path: /tmp/molecule/operator/molecule/api-test/graph.test.yaml:23
fatal: [localhost]: FAILED! => {
    "assertion": "api_namespaces_graph_json.elements.nodes | length == 2",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

**Comment by jmazzitelli** [comment]
Running this test with 1.38.1, the failure is different (but still fails only with OSSM 2.1 on OpenShift):

```
    TASK [Results: /namespaces/graph: https://kiali-istio-system.apps.aaa.com/api/namespaces/graph?namespaces=simple-mesh&graphType=workload&injectServiceNodes=false] ***
    task path: /tmp/molecule/operator/molecule/api-test/graph.test.yaml:16
    ok: [localhost] => {
        "msg": {
            "duration": 600,
            "elements": {
                "edges": [],
                "nodes": [
                    {
                        "data": {
                            "app": "simple-client",
                            "cluster": "Kubernetes",
                            "hasMissingSC": true,
                            "id": "371ae5a8eff043855792872141efa353",
                            "isIdle": true,
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "version": "v99",
                            "workload": "simple-client"
                        }
                    },
                    {
                        "data": {
                            "app": "simple-server",
                            "cluster": "Kubernetes",
                            "hasMissingSC": true,
                            "id": "440df82a974d31fc2ab0e4b7ba8bc1ed",
                            "isIdle": true,
                            "namespace": "simple-mesh",
                            "nodeType": "workload",
                            "version": "v99",
                            "workload": "simple-server"
                        }
                    }
                ]
            },
            "graphType": "workload",
            "timestamp": 1628545814
        }
    }
    
    TASK [Assert that we were able to get the graph info for the small 2-node mesh] ***
    task path: /tmp/molecule/operator/molecule/api-test/graph.test.yaml:23
fatal: [localhost]: FAILED! => {
    "assertion": "api_namespaces_graph_json.elements.edges | length == 1",
```

**Comment by jmazzitelli** [comment]
This failure is likely due to an OSSM 2.1 bug that is currently not fixed. Wait for a future QE build of 2.1 before testing this further.

**Comment by lucasponce** [comment]
Relabeled due https://github.com/kiali/kiali/issues/4246#issuecomment-896090320

**Comment by jshaughn** [comment]
Assigning this to @jmazzitelli for a retest now that 2.1 is released.

**Comment by jmazzitelli** [comment]
I just deployed the released version of OSSM 2.1 along with the released version of the Kiali operator. I deployed a 2.1 SMCP with kiali disabled, checked out the v1.36 branch of the operator (to get the branch's molecule tests) and then I ran this:

```
hack/run-molecule-tests.sh --cluster-type openshift --client-exe $(which oc) -dorp podman -at "api-test" -udsi true -oi skip -sv v1.36
```

The test passes. So things look Ok now.

```
                                api-test... success [9m 13s]
```
